### PR TITLE
MODE-2387 Added support for additional MapDB configuration options.

### PR DESCRIPTION
**Pull request by hchiorean**

### CODE REVIEW

File: modeshape-jcr/src/main/java/org/modeshape/jcr/index/local/LocalIndexProvider.java [hunk 1]
```diff
@@ -138,7 +148,33 @@ protected void doInitialize() throws RepositoryException {
                            file.getAbsolutePath());
         }
         // Get the database ...
-        this.db = DBMaker.newFileDB(file).make();
+        DBMaker<?> dbMaker = DBMaker.newFileDB(file);
+        if (this.cacheSize != null) {
+            dbMaker.cacheSize(cacheSize);
+            logger().trace("MapDB cache size set to {0}", cacheSize);
```
Comment (rhauch) [inline modeshape-jcr/src/main/java/org/modeshape/jcr/index/local/LocalIndexProvider.java]: What do you think about including the provider name in these messages?

**Comment by rhauch** [comment]
Looks great, especially if provider name added to debug messages (see above).

**Comment by hchiorean** [comment]
amended the commit

**Comment by rhauch** [comment]
Thanks. Looks great.

**Pull request merged**